lib/Entrypoint.js

```diff
@@ -2,24 +2,54 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-"use strict";
+/** @typedef {import("./Chunk")} Chunk */
 
 const ChunkGroup = require("./ChunkGroup");
-
+/**
+ *
+ * @description Entrypoint serves as an encapsulation primitive for chunks that are
+ * apart of a single ChunkGroup. They represent all bundles that need to be loaded for a
+ * single instance of a page. Multipage App Architectures will typically yeild multiple Entrypoint Objects
+ * inside of Compilation, where as a Single Page App, may only contain one with many lazy loaded chunks.
+ * @class Entrypoint
+ * @extends {ChunkGroup}
+ */
 class Entrypoint extends ChunkGroup {
+	/**
+	 * Creates an instance of Entrypoint.
+	 * @param {string} name the name of the entrypoint
+	 * @memberof Entrypoint
```

Comment:
do we really need the `memberof` directive?

Reply:
I kind of like this because when we get to a point to generate full reference documentation, we can use this as metadata. 

When using DocumentThis with vscode, it just auto generates for me. I don't see a problem with it personally especially when trying to reference type information from intellisense etc.

Reply:
Let's talk about this in #7122. The more directives we require, harder it gets to add typings and also, we should prefer inferred types over declared types because those update as code changes.

Reply:
I agree we can discuss there. If we need to change, I'll go in and remove where needed.